spec/system/publishers/publishers_can_select_a_job_application_for_interview_spec.rb

```diff
@@ -3,341 +3,350 @@
 
-          scenario "without an email" do
-            click_on I18n.t("buttons.save_and_continue")
-            expect(page).to have_content("Enter a valid email address")
-          end
-        end
-
-        context "with a received reference" do
-          before do
-            current_referee.reload
-            # simulate receipt of a reference
-            current_referee.job_reference.update!(reference_data.merge(updated_at: Date.tomorrow))
-            # remove previous emails so that current ones can be checked
-            ActionMailer::Base.deliveries.clear
-            current_referee.job_reference.mark_as_received
-            publisher_ats_interviewing_page.pre_interview_check_links.first.click
-          end
-
-          context "with a simple reference" do
-            let(:reference_data) { attributes_for(:job_reference, :reference_given) }
+          context "when the reference is declined" do
+            before do
+              current_referee.reload.job_reference.update!(attributes_for(:job_reference, :reference_declined).merge(updated_at: Date.tomorrow))
+              current_referee.reload.reference_request.update!(status: :received)
+            end
 
-            it "can progress to the page where the reference is shown" do
+            it "shows the reference as declined" do
+              publisher_ats_interviewing_page.pre_interview_check_links.first.click
               expect(publisher_ats_pre_interview_checks_page).to be_displayed
 
               publisher_ats_pre_interview_checks_page.reference_links.first.click
-              expect(publisher_ats_reference_request_page).to be_displayed
+              expect(page).to have_content("You will need to request a new referee")
+              expect(publisher_ats_pre_interview_checks_page.timeline).to have_content("Reference declined")
             end
+          end
 
-            it "send an email notification to the publisher that the reference had been received" do
-              expect(ActionMailer::Base.deliveries.map(&:to).flatten)
-                .to contain_exactly("publisher@contoso.com")
+          context "when the referee email is incorrect" do
+            before do
+              ActionMailer::Base.deliveries.clear
+              publisher_ats_interviewing_page.pre_interview_check_links.first.click
+              publisher_ats_pre_interview_checks_page.reference_links.first.click
+              within ".govuk-main-wrapper" do
+                within ".govuk-grid-column-two-thirds" do
+                  find("a.govuk-link").click
+                end
+              end
             end
 
-            context "when marking reference as complete" do
-              before do
-                publisher_ats_pre_interview_checks_page.reference_links.first.click
-                click_on "Mark as received"
-              end
+            let(:new_email) { Faker::Internet.email(domain: TEST_EMAIL_DOMAIN) }
+
+            scenario "with a good email" do
```

Comment:
"with a good email" nested inside "when the referee email is incorrect" is quite confusing to me. I think it means when the referee email is incorrect but then we replace it with a valid email?

Reply:
replaced with the word 'valid' instead - the other case was w/o an email at all